contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -1251,6 +1264,60 @@ SendColumnMetadataToken(int natts, bool sendRowStat)
 			}
 		}
 
+		//For Spatial Types
+			tup = SearchSysCache1(TYPEOID, ObjectIdGetDatum(col->pgTypeOid));  //Search SysCache for type name
+			typTup = (Form_pg_type) GETSTRUCT(tup);
+
+			/*
+			 * Check if it is spatial Data type
+			 * Send the Corresponding MetaData Columns
+			*/			
+			if(strcmp(NameStr(typTup->typname),"geometry") == 0 || strcmp(NameStr(typTup->typname),"geography") == 0)
```

Comment:
Instead of string matching, let's have a helper function in common extension which we will access through `pltsql_plugin_handler_ptr` rendezvous variable. Hint: check how `sqlvariant_get_pg_base_type` function has been implemented through the above rendezvous variable.

Reply:
used common_utility_plugin_ptr to access the helper function common extension just like other datatypes